report/wine_classification_report.ipynb

```diff
@@ -225,28 +267,134 @@
+   "metadata": {},
+   "source": [
+    "To find the optimal value of the hyperparameter `C` for our LR model that maximized the accuracy score, we performed a 10-fold cross-validated grid search with sklearn's `GridSearchCV`.\n",
+    "The `GridSearchCV` object was constructed using the following parameter grid 0.01, 0.1, 1, 10, 100, 1000 and a pipeline object made with sklearn's `make_pipeline` that implemented `StandardScaler` and `LogisticRegression` with `max_iter` of 2000. The `GridSearchCV` object was fit on the training set. The pickled object can be found here: `results/models/wine_pipeline.pickle`. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, the model was scored on the test set and the resulting `f1` and `accuracy` scores were used to assess model performance. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In terms of the model performance and its applicability to wine origin prediction, our current logistic regression model performs quite well on unseen test samples"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "![Accuracy scores for training and validation sets during hyperparameter optimization](../results/figures/wine_cv_C.png)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Figure 2: Accuracy scores for training and validation sets during hyperparameter optimization."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```{glue:figure} test_scores_df\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Table 1: Accuracy and F1 scores after model scoring on the test set."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Discussion"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "To further improve the classification accuracy, we may explore other models such as Support Vector Machines (SVM) and Random Forest to assess if they offer improved test accuracy. \n",
+    "In addition, diversifying our evaluation metrics can provide a more comprehensive understanding of our model's performance."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Limitations\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "  Metrics such as precision, recall, F1-score are good choices for imblanced class. According to our baseline model, the accuracy is of 40.33%, indicates that the most prevalent class occurs at a rate of 40%. This suggests a class imbalance (as we have three classes) "
```

Comment:
This is not really a limitation, something we're already accounting for in the model - limitations we can mention are maybe we're not sure how the model generalizes to Cultivars outside of the 3 we're classifying in the model

Reply:
We'll address in subsequent PR, gonna just merge this for now to get rid of conflicts